### PR TITLE
Added a devcontainer to make the contributing experience easier.

### PR DESCRIPTION
**Pull request by JoeRobich**
Borrowed the base of this from the Roslyn repo and tweaked as necessary.

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# See here for image contents: https://github.com/microsoft/vscode-dev-containers/tree/v0.192.0/containers/dotnet/.devcontainer/base.Dockerfile
+
+FROM mcr.microsoft.com/dotnet/sdk:8.0
+
+# Set up machine requirements to build the repo
+RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
+    && apt-get -y install --no-install-recommends curl git gnupg \
+    && curl -fsSL https://deb.nodesource.com/setup_20.x | bash - \
+    && apt-get install -y nodejs
```
Comment (JoeRobich) [inline .devcontainer/Dockerfile:9]: Installs NodeJS v20

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      "extensions": [
+        "ms-dotnettools.csharp",
+        "ms-dotnettools.csdevkit",
+        "EditorConfig.EditorConfig",
+        "ms-vscode.powershell",
+        "tintoy.msbuild-project-tools",
+        "ms-azure-devops.azure-pipelines",
+        "dbaeumer.vscode-eslint",
+        "esbenp.prettier-vscode",
+        "orta.vscode-jest"
```
Comment (JoeRobich) [inline .devcontainer/devcontainer.json:59]: Preinstalls the recommended extensions.

File: .devcontainer/devcontainer.json [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      ]
+    }
+  },
+  "postCreateCommand": "npm ci && npx gulp installDependencies"
```
Comment (JoeRobich) [inline .devcontainer/devcontainer.json:63]: Installs packages and dependencies

File: .devcontainer/devcontainer.json [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+          "*.slnf": "json",
+          "*.targets": "msbuild",
+          "*.vbproj": "msbuild",
+          "*.vsixmanifest": "xml",
+          "*.vstemplate": "xml",
+          "*.xlf": "xml",
+          "*.yml": "azure-pipelines"
+        },
+        // ms-dotnettools.csharp settings
+        "omnisharp.disableMSBuildDiagnosticWarning": true,
```
Comment (dibarbet) [inline .devcontainer/devcontainer.json:36]: probably can remove these O# options

File: .devcontainer/devcontainer.json [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        "omnisharp.enableEditorConfigSupport": true,
+        "omnisharp.enableImportCompletion": true,
+        "omnisharp.useModernNet": true,
+        "omnisharp.enableAsyncCompletion": true,
+        // ms-dotnettools.csdevkit settings
+        "dotnet.defaultSolution": "Roslyn.sln",
```
Comment (dibarbet) [inline .devcontainer/devcontainer.json:42]: and this

**Review by dibarbet — APPROVED** [review]

File: .devcontainer/devcontainer.json (hunk 5 above)
Comment (JoeRobich) [reply .devcontainer/devcontainer.json]: Removed!

**Pull request merged**